devicetypes/automateasia/wyfy-touch-zwave-switch.src/wyfy-touch-zwave-switch.groovy

```diff
@@ -0,0 +1,522 @@
+/**
+ *  WYFY TOUCH ZWAVE SWITCH
+ *
+ *  Copyright 2020 Edwin Tan Poh Heng
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+
+/*
+Description of State variables
+endpointCount: The number of endpoints
+endpointMap: A collection that maps childDeviceNetworkId to their endpoint 
+zwaveParemeterKey: A collection of the ZWave parameter's value and state
+*/
+
+metadata {
+	definition (name: "WYFY TOUCH ZWAVE SWITCH", namespace: "AutomateAsia", author: "Edwin Tan Poh Heng", cstHandler: true) {
```

Comment:
Please try to integrate your device with this DTH https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy
add `mnmn: "SmartThings, vid: "generic-switch"` to your fingerprints

Reply:
I can't integrate to this DTH [https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy](url).

Reason is because my device has specific configuration parameters that are unique to it.

Reply:
For CST devices we want to provide basic functionality with generic configuration, so please include your device in DTH I've mentioned above.

Reply:
May I ask, for the preferences related to WYFY Switch, how do you suggest that I include it into the generic DTH?

Reply:
Have included device into existing DTH as requested. See https://github.com/erocm123/SmartThingsPublic/pull/23